frontend/ui/src/pages/Channels/ConnectedChannelsList/ChannelListItem.tsx

```diff
@@ -0,0 +1,151 @@
+import {ReactComponent as GoogleLogo} from 'assets/images/icons/google_avatar.svg';
+import {ReactComponent as SMSLogo} from 'assets/images/icons/sms_avatar.svg';
+import {ReactComponent as WhatsappLogo} from 'assets/images/icons/whatsapp_avatar.svg';
+import {ReactComponent as AiryLogo} from 'assets/images/icons/airy_avatar.svg';
+import {ReactComponent as CheckMark} from 'assets/images/icons/checkmark.svg';
+import styles from './ChannelListItem.module.scss';
+import {RouteComponentProps, withRouter} from 'react-router-dom';
+import {CHANNELS_CHAT_PLUGIN_ROUTE, CHANNELS_FACEBOOK_ROUTE} from '../../../routes/routes';
+
+type ChannelItemProps = {
+  channel: Channel;
+} & ConnectedProps<typeof connector> &
+  RouteComponentProps<{channelId: string}>;
+
+const mapDispatchToProps = {
+  disconnectChannel,
+};
+
+const connector = connect(null, mapDispatchToProps);
+
+const ChannelItem = (props: ChannelItemProps) => {
+  const {channel} = props;
+  const [deletePopupVisible, setDeletePopupVisible] = useState(false);
+
+  const disconnectChannelRequestPayload = {
+    channelId: channel.id,
+  };
+
+  const channelIcon = (channel: Channel) => {
+    switch (channel.source) {
+      case ChannelSource.facebook:
+        if (channel?.metadata?.imageUrl) {
+          return <img className={styles.imageUrlLogo} src={channel.metadata.imageUrl} />;
+        } else {
+          return <FacebookLogo />;
+        }
+      case ChannelSource.google:
+        if (channel?.metadata?.imageUrl) {
+          return <img className={styles.imageUrlLogo} src={channel.metadata.imageUrl} />;
+        } else {
+          return <GoogleLogo />;
+        }
+      case ChannelSource.twilioSMS:
+        if (channel?.metadata?.imageUrl) {
+          return <img className={styles.imageUrlLogo} src={channel.metadata.imageUrl} />;
+        } else {
+          return <SMSLogo />;
+        }
+      case ChannelSource.twilioWhatsapp:
+        if (channel?.metadata?.imageUrl) {
+          return <img className={styles.imageUrlLogo} src={channel.metadata.imageUrl} />;
+        } else {
+          return <WhatsappLogo />;
+        }
+      case ChannelSource.chatPlugin:
+        if (channel?.metadata?.imageUrl) {
+          return <img className={styles.imageUrlLogo} src={channel.metadata.imageUrl} />;
+        } else {
+          return <AiryLogo />;
+        }
+      default:
+        return <AiryLogo />;
+    }
+  };
+
+  const editChannel = () => {
+    switch (channel.source) {
+      case ChannelSource.facebook:
+        return {pathname: CHANNELS_FACEBOOK_ROUTE + `/${channel.id}`, state: {channel: channel}};
+      case ChannelSource.google:
+        break;
+      case ChannelSource.twilioSMS:
+        break;
+      case ChannelSource.twilioWhatsapp:
+        break;
+      case ChannelSource.chatPlugin:
+        return {pathname: CHANNELS_CHAT_PLUGIN_ROUTE + `/${channel.id}`, state: {channel: channel}};
+    }
+  };
```

Comment:
the only thing changing here is going to be the path right? so we can/should make that clear by mapping channel names to paths (then this becomes one line of code)